apply/apply_Sobol.py

```diff
@@ -50,27 +56,29 @@
         y_sobol = evaluate(x_sobol, a)
         sa_sobol = analyze_sobol.analyze(problem, 
                     y_sobol, calc_second_order=False, num_resamples=1000, conf_level=0.95)
+
         # use toposort find parameter sa block
         conf_low = sa_sobol['total_rank_ci'][0]
         conf_up = sa_sobol['total_rank_ci'][1]
-
         abs_sort = partial_rank(len_params, conf_low, conf_up)
         rank_list = list(toposort(abs_sort))
 
         key = 'result_'+str(i)
         partial_order[key] = {j: list(rank_list[j]) for j in range(len(rank_list))}
-#         #save results of sobol indices
+
+        # Save results
         sa_total[key] = sa_sobol['ST']         
         sa_main[key] = sa_sobol['S1']
 
         error_dict[key], pool_res = group_fix(partial_order[key], evaluate, 
                         x_all, y_true, x_default, rand, pool_res, a, file_exist)
+
     with open(cache_file, 'w') as fp:
         json.dump(partial_order, fp, indent=2)
 else:
-    file_exist = True
     with open(cache_file, 'r') as fp:
         partial_order = json.load(fp)
+
     for key, value in partial_order.items():
         error_dict[key], pool_res = group_fix(value, evaluate, x_all, y_true, 
```

Comment:
This appears to do nothing when the cache file exists, so the calculation is done for no reason.

Is my understanding correct?